File: payola/web/client/src/main/scala/Layer.scala
package cz.payola.web.client

import s2js.adapters.js.dom.{CanvasRenderingContext2D, Canvas}

class Layer(val canvas: Canvas, val context: CanvasRenderingContext2D) {
    def setSize(size: Vector) {
        canvas.width = size.x;
        canvas.height = size.y;
    }
    
    def getSize: Vector = {
        Vector(canvas.width, canvas.height)
    }
}

File: payola/web/public/javascripts/cz/payola/web/client/Layer.js
goog.provide('cz.payola.web.client.Layer');
goog.require('cz.payola.web.client.Vector');
cz.payola.web.client.Layer = function(canvas, context) {
var self = this;
self.canvas = canvas;
self.context = context;
};
cz.payola.web.client.Layer.prototype.setSize = function(size) {
var self = this;
self.canvas.width = size.x;
self.canvas.height = size.y;
};
cz.payola.web.client.Layer.prototype.getSize = function() {
var self = this;
return new cz.payola.web.client.Vector(self.canvas.width, self.canvas.height);
};
cz.payola.web.client.Layer.prototype.metaClass_ = new s2js.MetaClass('cz.payola.web.client.Layer', []);
